File: frontend/src/lib/platformAdmin.ts
/**
 * Administrateurs plateforme EYWAI (accès Administration + RH complet).
 */

export type PlatformAdminUser = {
  is_platform_admin?: boolean;
  /** @deprecated compat API — préférer is_platform_admin */
  is_super_admin?: boolean;
  role?: string;
};

export function isPlatformAdmin(user: PlatformAdminUser | null | undefined): boolean {
  if (!user) return false;
  return Boolean(user.is_platform_admin || user.is_super_admin);
}

/** Rôle effectif côté RH : admin plateforme = admin entreprise. */
export function effectiveRhRole(user: PlatformAdminUser | null | undefined): string | undefined {
  if (!user) return undefined;
  if (isPlatformAdmin(user)) return 'admin';
  return user.role;
}

export function hasFullRhAccess(user: PlatformAdminUser | null | undefined): boolean {
  if (!user) return false;
  if (isPlatformAdmin(user)) return true;
  const role = user.role;
  return role === 'admin' || role === 'rh' || role === 'collaborateur_rh';
}

File: backend/app/core/platform_admin.py
"""
Helpers pour les administrateurs plateforme EYWAI (ex-super admins).

Un admin plateforme a accès à l'interface Administration et à l'interface RH
avec tous les droits sur toutes les entreprises.
"""

from __future__ import annotations

from typing import Any


def is_platform_admin(user: Any) -> bool:
    """True si l'utilisateur est admin plateforme (table super_admins active)."""
    return bool(
        getattr(user, "is_platform_admin", False)
        or getattr(user, "is_super_admin", False)
    )
